src/Core/Config/Configuration.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Friendica\Core\Config;
+
+/**
+ * This class is responsible for all system-wide configuration values in Friendica
+ * There are two types of storage
+ * - The Config-Files    (loaded into the FileCache @see Cache\IConfigCache )
+ * - The Config-DB-Table (per Config-DB-adapter @see Adapter\IConfigAdapter )
+ */
+class Configuration
+{
+	/**
+	 * @var Cache\IConfigCache
+	 */
+	private $configCache;
+
+	/**
+	 * @var Adapter\IConfigAdapter
+	 */
+	private $configAdapter;
+
+	/**
+	 * @param Cache\IConfigCache     $configCache   The configuration cache (based on the config-files)
+	 * @param Adapter\IConfigAdapter $configAdapter The configuration DB-backend
+	 */
+	public function __construct(Cache\IConfigCache $configCache, Adapter\IConfigAdapter $configAdapter)
+	{
+		$this->configCache = $configCache;
+		$this->configAdapter = $configAdapter;
+
+		$this->load();
+	}
+
+	/**
+	 * Returns the Config Cache
+	 *
+	 * @return Cache\IConfigCache
+	 */
+	public function getCache()
+	{
+		return $this->configCache;
+	}
+
+	/**
+	 * @brief Loads all configuration values of family into a cached storage.
+	 *
+	 * All configuration values of the system are stored in the cache ( @see IConfigCache )
+	 *
+	 * @param string $cat The category of the configuration value
+	 *
+	 * @return void
+	 */
+	public function load($cat = 'config')
+	{
+		// If not connected, do nothing
+		if (!$this->configAdapter->isConnected()) {
+			return;
+		}
+
+		// load the whole category out of the DB into the cache
+		$this->configCache->load($this->configAdapter->load($cat), true);
+	}
+
+	/**
+	 * @brief Get a particular user's config variable given the category name
+	 * ($cat) and a $key.
+	 *
+	 * Get a particular config value from the given category ($cat)
+	 * and the $key from a cached storage either from the $this->configAdapter
+	 * (@see IConfigAdapter ) or from the $this->configCache (@see IConfigCache ).
+	 *
+	 * @param string  $cat        The category of the configuration value
+	 * @param string  $key           The configuration key to query
+	 * @param mixed   $default_value optional, The value to return if key is not set (default: null)
+	 * @param boolean $refresh       optional, If true the config is loaded from the db and not from the cache (default: false)
+	 *
+	 * @return mixed Stored value or null if it does not exist
+	 */
+	public function get($cat, $key, $default_value = null, $refresh = false)
+	{
+		// if the value isn't loaded or refresh is needed, load it to the cache
+		if ($this->configAdapter->isConnected() &&
```

Comment:
@MrPetovan This one is the new "magic"

In other words (without the methods around it):
```php
if ($adapter->isConnected && ($adapter->in_db[$cat][$key] || $refresh) {}
```
This is will become the only check for each `Config::get()`, otherwise the Cache will be used. I hope that this statement is fast enough (or at least as fast as before).